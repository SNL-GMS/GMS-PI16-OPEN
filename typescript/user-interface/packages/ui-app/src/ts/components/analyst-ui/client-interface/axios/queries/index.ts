import * as EventQuery from './event-query';
import * as QcMaskQuery from './qc-mask-query';
import * as SignalDetectionQuery from './signal-detection-query';
import * as WaveformQuery from './waveform-query';
import * as WorkflowIntervalQuery from './workflow-interval-query';
import * as WorkflowQuery from './workflow-query';
import * as WorkflowQueryUtil from './workflow-query-util';

export {
  EventQuery,
  QcMaskQuery,
  SignalDetectionQuery,
  WaveformQuery,
  WorkflowIntervalQuery,
  WorkflowQuery,
  WorkflowQueryUtil
};
