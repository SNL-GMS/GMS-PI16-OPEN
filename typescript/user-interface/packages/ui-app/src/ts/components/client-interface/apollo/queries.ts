import { ConfigurationQueries } from '@gms/common-graphql';
import { graphql } from 'react-apollo';

// ----- UI Configuration Queries ------

/**
 * Returns a wrapped component providing the `UIConfiguration` query.
 *
 * @export
 * @template T defines the component base props required
 * @returns the wrapped component
 */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function graphqlUIConfigurationQuery() {
  return graphql(ConfigurationQueries.uiConfigurationQuery, { name: 'uiConfigurationQuery' });
}
