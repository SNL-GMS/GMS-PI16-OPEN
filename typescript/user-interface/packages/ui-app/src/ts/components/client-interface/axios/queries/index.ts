import * as ProcessingAnalystConfigurationQuery from './processing-analyst-configuration-query';
import * as ProcessingCommonConfigurationQuery from './processing-common-configuration-query';
import * as OperationalTimePeriodConfigurationQuery from './processing-operational-time-period-configuration-query';
import * as ProcessingStationGroupNamesConfigurationQuery from './processing-station-group-names-configuration-query';
import * as QueryUtils from './query-util';
import * as StationGroupQuery from './station-group-query';
import * as StationGroupsDefinitionQuery from './station-groups-definition-query';
import * as StationQueryUtil from './station-query-util';
import * as StationsChangeTimesQuery from './stations-change-times-query';
import * as StationsDefinitionQuery from './stations-definition-query';
import * as SystemMessageDefinitionQuery from './system-message-definition-query';
import * as UserProfileQuery from './user-profile-query';

export {
  OperationalTimePeriodConfigurationQuery,
  ProcessingAnalystConfigurationQuery,
  ProcessingCommonConfigurationQuery,
  ProcessingStationGroupNamesConfigurationQuery,
  QueryUtils,
  StationGroupQuery,
  StationGroupsDefinitionQuery,
  StationQueryUtil,
  StationsChangeTimesQuery,
  StationsDefinitionQuery,
  SystemMessageDefinitionQuery,
  UserProfileQuery
};
