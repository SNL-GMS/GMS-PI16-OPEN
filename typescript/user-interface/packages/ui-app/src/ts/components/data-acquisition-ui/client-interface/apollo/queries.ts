import { ProcessingStationQueries } from '@gms/common-graphql';
import { graphql } from 'react-apollo';

// ----- Station Processing Queries ------

/**
 * Returns a wrapped component providing the `defaultStationsQuery` query.
 *
 * @export
 * @returns the wrapped component
 */
export function graphqlDefaultProcessingStationsQuery() {
  return graphql(ProcessingStationQueries.defaultProcessingStationsQuery, {
    name: 'defaultStationsQuery'
  });
}
