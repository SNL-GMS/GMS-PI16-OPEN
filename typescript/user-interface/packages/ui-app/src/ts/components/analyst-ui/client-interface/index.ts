// import * as IanMutations from './axios/mutations';
import * as IanMutations from './axios/mutations';
import * as IanQueries from './axios/queries';
import * as IanQueryAndMutationTypes from './axios/types';

export { IanQueries };
export { IanMutations };
export { IanQueryAndMutationTypes };
