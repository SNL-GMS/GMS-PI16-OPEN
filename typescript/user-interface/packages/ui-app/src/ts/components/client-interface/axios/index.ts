import * as Mutations from './mutations';
import * as Queries from './queries';
import * as Subscriptions from './subscriptions';

export { Queries };
export { Mutations };
export { Subscriptions };
