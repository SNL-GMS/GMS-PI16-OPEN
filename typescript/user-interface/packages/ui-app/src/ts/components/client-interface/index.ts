import * as Mutations from './axios/mutations';
import * as Queries from './axios/queries';
import * as Subscriptions from './axios/subscriptions';
import * as QueryAndMutationTypes from './axios/types';

export * from './axios/queries/query-util';
export { Queries };
export * from './axios/mutations/mutation-util';
export { Mutations };
export { Subscriptions };
export * from './axios/user-profile-util';
export { QueryAndMutationTypes };
