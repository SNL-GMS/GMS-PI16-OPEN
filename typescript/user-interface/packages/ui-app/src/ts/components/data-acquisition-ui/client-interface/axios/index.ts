import * as Queries from './queries';

export { Queries };
