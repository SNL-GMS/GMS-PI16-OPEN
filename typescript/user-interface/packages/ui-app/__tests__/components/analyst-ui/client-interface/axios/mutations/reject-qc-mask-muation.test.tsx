/* eslint-disable jest/expect-expect */
import { QcMaskTypes } from '@gms/common-model';
import Axios from 'axios';
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { IanMutations } from '../../../../../../src/ts/components/analyst-ui/client-interface';
import { QueryUtils } from '../../../../../../src/ts/components/client-interface/axios/queries';
import {
  expectThatCompositionInjectsAProp,
  expectThatMutationHookMakesAxiosCall
} from '../../../../../utils/query-mutation-test-utils';

// eslint-disable-next-line no-console
console.error = jest.fn();
Date.now = jest.fn().mockReturnValue(() => 1000);

const success = 'success';
Axios.request = jest.fn().mockReturnValue(Promise.resolve(success));

describe('Reject QcMask mutation', () => {
  it('reject qcMaskMutationConfig is defined', () => {
    expect(IanMutations.RejectQcMaskMutation.qcMaskMutationConfig).toBeDefined();
  });

  const mutationArgs: QcMaskTypes.RejectQcMaskMutationArgs = {
    maskId: 'foo',
    inputRationale: 'foo'
  };

  it('qcMaskMutationConfig on success invalidates the qcMask query to force a re-fetch', async () => {
    QueryUtils.queryCache.invalidateQueries = jest.fn();

    const prevNumCalls = (QueryUtils.queryCache.invalidateQueries as jest.Mock).mock.calls.length;
    await IanMutations.RejectQcMaskMutation.qcMaskMutationConfig.onSuccess();
    const { calls } = (QueryUtils.queryCache.invalidateQueries as jest.Mock).mock;
    // check for the second argument of the call, to make sure it was called with the newly provided data.
    expect(calls).toHaveLength(prevNumCalls + 1);
  });

  it('rejectQcMasks makes a call to the server with the expected data', () => {
    // eslint-disable-next-line @typescript-eslint/no-floating-promises, jest/valid-expect-in-promise, jest/valid-expect
    expect(IanMutations.RejectQcMaskMutation.rejectQcMasks(mutationArgs))
      .resolves.toBe(success)
      // eslint-disable-next-line no-console
      .catch(e => console.error(e));
  });

  it('rejects a wrapped component that matches a snapshot', () => {
    // eslint-disable-next-line @typescript-eslint/ban-types
    const TestComponent: React.FC<{}> = () => <div>test</div>;
    const wrapperFn = IanMutations.RejectQcMaskMutation.withRejectQcMaskMutation();
    const WrappedComponent = wrapperFn(TestComponent);
    const wrapper = Enzyme.mount(<WrappedComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can inject a rejectQcMask mutation prop into a wrapped component', () => {
    expectThatCompositionInjectsAProp(
      IanMutations.RejectQcMaskMutation.withRejectQcMaskMutation,
      'rejectQcMasks'
    );
  });

  it('makes a request to the server when the qcMask mutation is called', () => {
    expectThatMutationHookMakesAxiosCall(
      IanMutations.RejectQcMaskMutation.useQcMaskMutation,
      mutationArgs
    );
  });
});
