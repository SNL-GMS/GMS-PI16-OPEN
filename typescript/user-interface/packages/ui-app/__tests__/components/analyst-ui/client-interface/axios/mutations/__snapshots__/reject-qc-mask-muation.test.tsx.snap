// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reject QcMask mutation rejects a wrapped component that matches a snapshot 1`] = `
<Component>
  <ComponentWithMutation>
    <TestComponent
      rejectQcMasks={[Function]}
    >
      <div>
        test
      </div>
    </TestComponent>
  </ComponentWithMutation>
</Component>
`;
